AK/String.h

```diff
@@ -144,12 +146,15 @@ class String {
     bool is_empty() const { return length() == 0; }
     size_t length() const { return m_impl ? m_impl->length() : 0; }
     const char* characters() const { return m_impl ? m_impl->characters() : nullptr; }
-    char operator[](size_t i) const
+    const char& operator[](size_t i) const
     {
-        ASSERT(m_impl);
+        ASSERT(i < length());
```

Comment:
Why are we removing the `m_impl` assertion?
`StringImpl::operator[]` already asserts that `i` is within range.
However, if `m_impl` is null, it would be an error to dereference it.

Reply:
`RefPtr::operator*` asserts that `m_impl` is not null.

Reply:
Oh, that's true! Good point :)